command/root.go

```diff
@@ -250,6 +250,30 @@ func determineBaseRepo(cmd *cobra.Command, ctx context.Context) (ghrepo.Interfac
 	return baseRepo, nil
 }
 
+func convertRepoInterfaceToRepository(ctx context.Context, repo ghrepo.Interface) (*api.Repository, error) {
+	apiClient, err := apiClientForContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	remotes, err := ctx.Remotes()
+	if err != nil {
+		return nil, err
+	}
+
+	repoContext, err := context.ResolveRemotesToRepos(remotes, apiClient, "")
```

Comment:
I think this method of resolving is too complex for this use case and that resolving a "ghrepo" to a `*Repository` object could be done simpler. Additionally, it doesn't look like the `repo` argument to the parent function is respected in the lookup.

We have a function that can resolve a ghrepo to a Repository: `api.GitHubRepo()`. Currently, the payload returned does _not_ include the information about the default branch, but you could expand the fields listed in that query to include `defaultBranchRef { name }`.

Reply:
One more optimization that we could easily do: since the `repo ghrepo.Interface` object might already be an `*api.Repository`, we could detect and return that and avoid making an API request:

```go
if r, ok := repo.(*api.Repository); ok && r.DefaultBranchRef.Name != "" {
  return r, nil
}
return api.GitHubRepo(repo)
```

Reply:
I got rid of the `convertRepoInterfaceToRepository` method and replaced it with `api.GitHubRepo`. I'm against the type checking idea though. It feels like extra code without much benefit. But I'm interested in exploring how to ensure we always get an api.Repository (or why we sometimes only need a ghrepo interface)